tests/python/test_api.py

```diff
@@ -0,0 +1,89 @@
+    'hex_to_rgb', 'i', 'i16', 'i32', 'i64', 'i8', 'ij', 'ijk', 'ijkl', 'ijl',
+    'ik', 'ikl', 'il', 'init', 'int16', 'int32', 'int64', 'int8', 'is_active',
+    'is_logging_effective', 'j', 'jk', 'jkl', 'jl', 'k', 'kernel', 'kl', 'l',
+    'lang', 'length', 'linalg', 'log', 'max', 'mesh_local', 'mesh_patch_idx',
+    'metal', 'min', 'ndarray', 'ndrange', 'no_activate', 'one', 'opengl',
+    'parallelize', 'polar_decompose', 'pow', 'profiler', 'randn', 'random',
+    'raw_div', 'raw_mod', 'rescale_index', 'reset', 'rgb_to_hex', 'root',
+    'round', 'rsqrt', 'select', 'set_logging_level', 'sin',
+    'sparse_matrix_builder', 'sqrt', 'static', 'static_assert', 'static_print',
+    'stop_grad', 'svd', 'sym_eig', 'sync', 'tan', 'tanh', 'template', 'tools',
+    'types', 'u16', 'u32', 'u64', 'u8', 'ui', 'uint16', 'uint32', 'uint64',
+    'uint8', 'vulkan', 'wasm', 'x64', 'x86_64', 'zero'
+]
+user_api[ti.Field] = [
+    'copy_from', 'dtype', 'fill', 'from_numpy', 'from_torch', 'parent',
+    'shape', 'snode', 'to_numpy', 'to_torch'
+]
+user_api[ti.FieldsBuilder] = [
+    'bit_array', 'bit_struct', 'bitmasked', 'deactivate_all', 'dense',
+    'dynamic', 'finalize', 'lazy_grad', 'place', 'pointer'
+]
+user_api[ti.Matrix] = [
+    'all', 'any', 'cast', 'cols', 'cross', 'determinant', 'diag', 'dot',
+    'field', 'fill', 'identity', 'inverse', 'max', 'min', 'ndarray', 'norm',
+    'norm_inv', 'norm_sqr', 'normalized', 'one', 'outer_product', 'rotation2d',
+    'rows', 'sum', 'to_list', 'to_numpy', 'trace', 'transpose', 'unit', 'w',
+    'x', 'y', 'z', 'zero'
+]
+user_api[ti.MatrixField] = [
+    'copy_from', 'dtype', 'fill', 'from_numpy', 'from_torch',
+    'get_scalar_field', 'parent', 'shape', 'snode', 'to_numpy', 'to_torch'
+]
+user_api[ti.MatrixNdarray] = [
+    'copy_from', 'element_shape', 'fill', 'from_numpy', 'to_numpy'
+]
+user_api[ti.Ndarray] = ['copy_from', 'element_shape', 'fill']
+user_api[ti.SNode] = [
+    'bit_array', 'bit_struct', 'bitmasked', 'deactivate_all', 'dense',
+    'dynamic', 'lazy_grad', 'parent', 'place', 'pointer', 'shape'
+]
+user_api[ti.ScalarField] = [
+    'copy_from', 'dtype', 'fill', 'from_numpy', 'from_torch', 'parent',
+    'shape', 'snode', 'to_numpy', 'to_torch'
+]
+user_api[ti.ScalarNdarray] = [
+    'copy_from', 'element_shape', 'fill', 'from_numpy', 'to_numpy'
+]
+user_api[ti.Struct] = ['field', 'fill', 'items', 'keys', 'to_dict']
+user_api[ti.StructField] = [
+    'copy_from', 'dtype', 'fill', 'from_numpy', 'from_torch',
+    'get_member_field', 'keys', 'parent', 'shape', 'snode', 'to_numpy',
+    'to_torch'
+]
+user_api[ti.VectorNdarray] = [
+    'copy_from', 'element_shape', 'fill', 'from_numpy', 'to_numpy'
+]
+
+
+@pytest.mark.parametrize('src', user_api.keys())
+@test_utils.test(arch=ti.cpu)
+def test_api(src):
+    # When Python version is below 3.7, deprecated names are
+    # handled as normal names, which will fail this test.
+    assert sys.version_info < (3, 7) or [
```

Comment:
We might need a better error message to instruct developers how to tell if their apis should be added or just hidden, and how to update the lists if they really want to expose ;) 
But this can be a followup for sure!